src/SmartComponents/BaselinesPage/EditBaseline/helpers.js

```diff
@@ -272,25 +302,57 @@ function filterBaselineData(baselineData, expandedRows) {
 }
 
 function toggleExpandedRow(expandedRows, factName) {
+    let newExpandedRows;
+
     if (expandedRows.includes(factName)) {
-        expandedRows = expandedRows.filter(fact => fact !== factName);
+        newExpandedRows = expandedRows.filter(fact => fact !== factName);
     } else {
-        expandedRows.push(factName);
+        newExpandedRows = expandedRows.slice();
+        newExpandedRows.splice(0, 0, factName);
+    }
+
+    return newExpandedRows;
+}
+
+function findIsSelected(factData, id, isSelected) {
+    let allSubFactsSelected = true;
+
+    for (let i = 0; i < factData.length; i++) {
+        if (factData[i][0] === id && isSelected && !factData[i].selected) {
+            allSubFactsSelected = true;
+        } else if (!factData[i].selected) {
+            return false;
```

Comment:
The function makes me think it will always return an array of selected values. This return here exits the for loop and would give me back a boolean breaking my expectation.